pkg/collector/corechecks/servicediscovery/module/comm_test.go

```diff
@@ -67,35 +63,88 @@ func TestIgnoreCommsLengths(t *testing.T) {
 	}
 }
 
-// buildLongProc returns the path of a symbolic link to a binary file
-func buildLongProc(t *testing.T) string {
+// TestIgnoreKernelThread check that kernel thread is ignored
+func TestIgnoreKernelThread(t *testing.T) {
+	proc := &process.Process{
+		Pid: 1,
+	}
+	// Pid=1 is not kernel thread and should not be ignored
+	ignore := shouldIgnoreKernelThread(proc)
+	require.Equal(t, ignore, false)
+
+	// Pid=2 is kernel thread [kthreadd] and should be ignored
+	proc.Pid = 2
+	ignore = shouldIgnoreKernelThread(proc)
+	require.Equal(t, ignore, true)
```

Comment:
The code "ignores" pid `2` just because it does not exist, regardless of if it is a kernel thread

Reply:
I assumed that in most cases Pid=1. (init) and Pid=2 (kernel thread daemon) exist on the system. However this test is irrelevant, I removed it